Description: Thunderbird doesn't unescape the attachment URI's
 correctly, so just convert them back to filenames again
Author: Chris Coulson <chris.coulson@canonical.com>
Bug-Ubuntu: https://launchpad.net/bugs/757976
Forwarded: no

--- nemo-sendto-3.0.0.orig/src/plugins/evolution/evolution.c
+++ nemo-sendto-3.0.0/src/plugins/evolution/evolution.c
@@ -277,6 +277,7 @@ static void
 get_thunderbird_mailto (GtkWidget *contact_widget, GString *mailto, GList *file_list)
 {
 	GList *l;
+	gchar *filename;
 
 	g_string_append (mailto, "-compose \"");
 	if (email != NULL) {
@@ -291,9 +292,17 @@ get_thunderbird_mailto (GtkWidget *conta
 		if (text != NULL && *text != '\0')
 			g_string_append_printf (mailto, "to='%s',", text);
 	}
-	g_string_append_printf (mailto,"attachment='%s", (char *)file_list->data);
+
+	/* XXX: Work around https://launchpad.net/bugs/757976 . Thunderbird
+	 * doesn't properly unescape the file URI's, so just convert them back
+	 * to filenames again */
+	filename = g_filename_from_uri ((char *)file_list->data, NULL, NULL);
+	g_string_append_printf (mailto,"attachment='%s", filename);
+	g_free (filename);
 	for (l = file_list->next ; l; l=l->next){
-		g_string_append_printf (mailto,",%s", (char *)l->data);
+		filename = g_filename_from_uri ((char *)l->data, NULL, NULL);
+		g_string_append_printf (mailto,",%s", filename);
+		g_free (filename);
 	}
 	g_string_append (mailto, "'\"");
 }
